src/modm/math/matrix.hpp

```diff
@@ -48,33 +47,40 @@ namespace modm
 	 * \author	Niklas Hauser
 	 * \author	Fabian Greif
 	 */
-	template<typename T, uint8_t ROWS, uint8_t COLUMNS>
+	template<typename T, std::size_t ROWS, std::size_t COLUMNS>
 	class Matrix
 	{
 	public:
+		static constexpr std::size_t RowCount = ROWS;
+		static constexpr std::size_t ColumnCount = COLUMNS;
+		static constexpr std::size_t ElementCount = RowCount * ColumnCount;
+
+		T element[ElementCount];
+
 		/**
 		 * \brief	Default Constructor
 		 *
-		 * Creates a Matrix with uninitialized elements. Use zeroMatrix() to
-		 * create a matrix with all elements set to zero.
+		 * Creates a Matrix with uninitialized ElementCount. Use zeroMatrix() to
+		 * create a matrix with all ElementCount set to zero.
 		 */
-		Matrix();
+		constexpr Matrix() = default;
 
 		/**
 		 * \brief	Create a matrix from an array
 		 *
 		 * Example:
 		 * \code
-		 * const int16_t m[6] = {
+		 *
+		 * modm::Matrix<int16_t, 3, 2> a({
 		 *     1, 2,
 		 *     3, 4,
 		 *     5, 6,
-		 * };
-		 *
-		 * modm::Matrix<int16_t, 3, 2> a(m);
+		 * });
 		 * \endcode
 		 */
-		Matrix(const T *data);
+		constexpr Matrix(const T *data) {
```

Comment:
That constructor is dangerous, it takes an array as a raw pointer without any size information. I would change this to `std::span<const T, ElementCount>`. Then users have to be explicit when doing something nasty.